controller_manager:
  ros__parameters:
    update_rate: 200  # Hz

    crane_x7_arm_controller:
      type: joint_trajectory_controller/JointTrajectoryController
    crane_x7_gripper_controller:
      type: position_controllers/GripperActionController
    joint_state_controller:
      type: joint_state_broadcaster/JointStateBroadcaster

crane_x7_arm_controller:
  ros__parameters:
    joints:
      - crane_x7_shoulder_fixed_part_pan_joint
      - crane_x7_shoulder_revolute_part_tilt_joint
      - crane_x7_upper_arm_revolute_part_twist_joint
      - crane_x7_upper_arm_revolute_part_rotate_joint
      - crane_x7_lower_arm_fixed_part_joint
      - crane_x7_lower_arm_revolute_part_joint
      - crane_x7_wrist_joint

    command_interfaces:
      - position

    state_interfaces:
      - position

crane_x7_gripper_controller:
  ros__parameters:
    joint: crane_x7_gripper_finger_a_joint

    command_interfaces:
      - position

    state_interfaces:
      - position